js/main.js

```diff
@@ -126,13 +174,150 @@ const showBigPicture = (photo) => {
+
+    if (imgUpload.classList.contains('effects__preview--none')) {
+      effectLevel.classList.add('hidden');
+    } else {
+      effectLevel.classList.remove('hidden');
+    }
+  }
+};
 
+uploadForm.addEventListener('change', onEffectChange);
+
+const setGrayscale = () => `filter: grayscale(${effectLevelValue.value / Effect.SATURATION_MAX})`;
+const setSepia = () => `filter: sepia(${effectLevelValue.value / Effect.SATURATION_MAX})`;
+const setInvert = () => `filter: invert(${effectLevelValue.value}%)`;
+const setBlur = () => `filter: blur(${Math.round(effectLevelValue.value / Effect.SATURATION_STEP_SPECIAL)}px)`;
+const setBrightness = () => `filter: brightness(${Math.round(effectLevelValue.value / Effect.SATURATION_STEP_SPECIAL)})`;
+const setOriginal = () => '';
+
+const saturationFilterMap = new Map([
+  ["effects__preview--chrome", setGrayscale()],
+  ["effects__preview--sepia", setSepia()],
+  ["effects__preview--marvin", setInvert()],
+  ["effects__preview--phobos", setBlur()],
+  ["effects__preview--heat", setBrightness()],
+  ["effects__preview--none", setOriginal()],
+]);
+
+const onSaturationChange = () => {
+  const [saturationKey] = Array.from(imgUpload.classList);
+  imgUpload.style = saturationFilterMap.get(saturationKey);
+};
+
+effectLevelPin.addEventListener('mouseup', onSaturationChange);
+
+const getHashtagTooShortMessage = (hashtag) => `Ещё минимум ${Hashtag.MIN_LENGTH - hashtag.length} симв.`;
+
+const getHashtagTooLongMessage = (hashtag) => `Удалите лишние ${hashtag.length - Hashtag.MAX_LENGTH} симв.`;
+
+const getTooManyHashtagsMessage = () => `Можно ввести только ${Hashtag.MAX_COUNT} хэштегов`;
+
+const getHashtagDuplicateMessage = (hashtag) => `Вы ввели ${hashtag} хэштег более одного раза`;
+
+const setHashtagValidationMessage = (hashtag, index, hashtags) => {
+  const [firstLetter] = hashtag;
+  if (firstLetter !== '' && firstLetter !== '#') {
+    hashtagInput.setCustomValidity(HashtagValidationMessage.FIRST_LETTER_INVALID);
+  } else if (hashtag.length < Hashtag.MIN_LENGTH) {
+    hashtagInput.setCustomValidity(getHashtagTooShortMessage(hashtag));
+  } else if (hashtag.length > Hashtag.MAX_LENGTH) {
+    hashtagInput.setCustomValidity(getHashtagTooLongMessage(hashtag));
+  } else if (!Hashtag.REGEX.test(hashtag)) {
+    hashtagInput.setCustomValidity(HashtagValidationMessage.SPECIAL_CHARACTER);
+  } else if (index >= Hashtag.MAX_COUNT) {
+    hashtagInput.setCustomValidity(getTooManyHashtagsMessage());
+  } else if (hashtags.some((element, innerIndex) => element.toLowerCase() === hashtag.toLowerCase() && element[innerIndex] !== hashtag[index])) {
+    hashtagInput.setCustomValidity(getHashtagDuplicateMessage(hashtag));
+  } else {
+    hashtagInput.setCustomValidity('');
+  }
+
+  hashtagInput.reportValidity();
+};
+
+const onHashtagInput = () => {
+  const hashtags = hashtagInput.value.split(' ');
+
+  hashtags.forEach((hashtag, index, arr) => setHashtagValidationMessage(hashtag, index, arr));
```

Comment:
```suggestion
  hashtags.forEach(setHashtagValidationMessage);
```

зачем ты создаешь функцию коллбек, что имеет ту же сигнатуру, что и функция вызываемая внутри этой функции? Передай ссыку на нее и все.